File: Problems/Mathematics/Day-01/sol/Arnav_Singh/Solution1.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() {
     int t;
    cin >> t;
    while(t--) {
        int a,b,c;
        cin>>a>>b>>c;
        if((abs(b-a))*2<max(a,max(b,c))){
            cout << -1 <<'\n';
            continue;
        }
        if(c+(abs(b-a))>(abs(b-a)*2)){
            cout<<(c+(abs(b-a)))%(abs(b-a)*2)<<'\n';
        }
        else{
            cout<<c+(abs(b-a))<<'\n';
        }
    }
    
    return 0;
}

File: Problems/Mathematics/Day-01/sol/VaradSinghal/solution1.py
def solve():
    t = int(input())
    for _ in range(t):
        a, b, c = map(int, input().split())
        
        d = abs(a - b)
        n = 2 * d
        
        if n == 0 or max(a, b, c) > n:
            print(-1)
            continue
        
        half = n // 2
 
        if c + half <= n:
            print(c + half)
        else:
            print(c - half)

# Run
solve()
